pkg/sql/pgwire/pgwirebase/too_big_error.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package pgwirebase
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/cockroachdb/errors"
+	"github.com/gogo/protobuf/proto"
+)
+
+// withMessageTooBig decorates an error when a read overflows the ReadBuffer.
+type withMessageTooBig struct {
+	cause error
+	size  int
+}
+
+var _ error = (*withMessageTooBig)(nil)
+var _ errors.SafeDetailer = (*withMessageTooBig)(nil)
+var _ fmt.Formatter = (*withMessageTooBig)(nil)
+var _ errors.Formatter = (*withMessageTooBig)(nil)
+
+func (w *withMessageTooBig) Error() string         { return w.cause.Error() }
+func (w *withMessageTooBig) Cause() error          { return w.cause }
+func (w *withMessageTooBig) Unwrap() error         { return w.cause }
+func (w *withMessageTooBig) SafeDetails() []string { return []string{strconv.Itoa(w.size)} }
+
+func (w *withMessageTooBig) Format(s fmt.State, verb rune) { errors.FormatError(w, s, verb) }
+
+func (w *withMessageTooBig) FormatError(p errors.Printer) (next error) {
```

Comment:
`FormatError` and the `errors.Formatter` interface are not necessary for trivial wrappers -- the formatting code already knows how to deal with those.